classes/publication/Repository.php

```diff
@@ -228,19 +228,20 @@ public function edit(Publication $publication, array $params)
 
         parent::edit($publication, $params);
 
-        // Create or delete the thumbnail of a cover image
-        if (array_key_exists('coverImage', $params)) {
-            $publicFileManager = new PublicFileManager();
-            $submission = Repo::submission()->get($publication->getData('submissionId'));
+        $updatedPublication = $this->get($publication->getId());
+        $coverImages = $updatedPublication->getData('coverImage');
 
+        // Create or delete the thumbnail of a cover image
+        $publicFileManager = new PublicFileManager();
+        $submission = Repo::submission()->get($publication->getData('submissionId'));
+        if ($coverImages) {
             // Get the submission context
             $submissionContext = $this->request->getContext();
             if ($submissionContext->getId() !== $submission->getData('contextId')) {
                 $submissionContext = Services::get('context')->get($submission->getData('contextId'));
             }
 
-            foreach ($params['coverImage'] as $localeKey => $newCoverImage) {
-
+            foreach ($coverImages as $localeKey => $newCoverImage) {
```

Comment:
Here I am not sure, but: 
What if this function is called from somewhere else in the code, not when changing the cover image, so that $oldCoverImage and $coverImages are the same? -- It seems that the thumbnails will be re-created...
That's why I thought better to consider `$params['coverImage']`... -- this should exist only when the form for changing cover images is used... 
Right?

Reply:
@bozana the thumbnail will be created again either way here. But, I agree that we need to take into account the `params` array. I changed the code to a more-or-less the same code that you are proposing, with a small change in order not to recreate the thumbnail, unless a new image is uploaded.